teps/0053-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,477 @@
+
+### Ownership
+
+The ownership is controlled through the use of OWNERS file that allows
+the user listed in the file to approve PR to the sub-tree. e.g. OWNERS
+in the root directory controls the entire catalog while OWNERS file
+placed at /{resource-type}/{resource-name}/OWNERS is allowed to
+approve changes to all versions of the {resource-name}.
+
+- The root OWNERS file maps the tektoncd/catalog-maintainers
+  team. This team is responsible for the overall catalog, from
+  maintaining Resources (especially those who do not have a specific
+  set of OWNERS), making sure the testing infrastructure is healthy,
+  triaging issues, …
+
+- The *per-resource*OWNERS are responsible for maintaining their
+  Resource (updating images, behaviour, …)
+
+### Organization
+
+```bash=
+# Optional: owner(s) and reviewer(s) of the resource (all versions)
+./{resource-type}/{resource-name}/OWNERS
+# The README of the resource (what is the resource about, list of versions, …)
+./{resource-type}/{resource-name}/README.md
+
+# The resource itself
+./{resource-type}/{resource-name}/{version}/{resource}.yaml
+
+# The README of the versioned resource (usage, …)
+./{resource-type}/{resource-name}/{version}/README.md
+
+# The install script which will install all the related resources
+./{resource-type}/{resource-name}/{version}/install-script.sh
+
+# Tasks directory which refers to the task used in the pipeline
+./{resource-type}/{resource-name}/{version}/tasks/..
+
+# Optional: Test working samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/tests/…
+# Optional: Addition samples.
+# Those samples would be used to run automated tests
+./{resource-type}/{resource-name}/{version}/samples/…
+```
+
+For example:
+
+```bash=
+
+./pipeline/                     👈 The kind of the resource
+
+    /p1                         👈 Definition file must have same name
+       /OWNERS                  👈 owners of this resource
+       /0.1
+         /README.md
+         /install-script.sh     👈 To install all resources related to pipeline
+         /p1.yaml               👈 The file name should match the resource name
+         /samples/
+         /tasks                 👈 Tasks which pipelines refers
+         /support
+       /0.2/...
+
+    /p2
+       /OWNERS
+       /README.md
+       /0.1
+         /README.md
+         /p2.yaml
+         /samples/
+         /tasks
+```
+
+#### Fields added to the yaml file
+
+```yaml=
+metadata:
+  name: <pipeline-name>                           👈 MUST: Name of file and pipeline name should be the same
+  labels:
+    app.kubernetes.io/version: "0.1"              👈 MUST: version of the resource
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"     👈 MUST: version of pipeline
+    tekton.dev/category: ""                      👈 MUST: Category of a resource
+    tekton.dev/tags: <tags>                       👈 Optional: Comma separated list of tags
```

Comment:
what is the difference between category and tag?

Reply:
Initially all tags were mapped to categories in Hub, for e.g. [config.yaml](https://github.com/tektoncd/hub/blob/master/config.yaml) so whenever a new tag was added in a task it was mapped to a category called `others`. So before every release we had to manually map these new tags to some category, hence after the discussion in Catalog and Hub WG, we created a proposal for adding a category as an annotation. So Hub would now filter the resources based on categories and tags would work as keys for searching a resource. We have a [PR](https://github.com/tektoncd/community/pull/346) to update the [TEP 0003- Tekton Catalog Organization](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md)